agent/config/runtime.go

```diff
@@ -48,160 +55,1173 @@ type RuntimeConfig struct {
-	CheckUpdateInterval         time.Duration
-	Checks                      []*structs.CheckDefinition
-	ClientAddrs                 []*net.IPAddr
-	DNSAddrs                    []net.Addr
-	DNSPort                     int
-	DataDir                     string
-	DevMode                     bool
-	DisableAnonymousSignature   bool
-	DisableCoordinates          bool
-	DisableHostNodeID           bool
-	DisableKeyringFile          bool
-	DisableRemoteExec           bool
-	DisableUpdateCheck          bool
-	DiscardCheckOutput          bool
-	EnableACLReplication        bool
-	EnableDebug                 bool
-	EnableScriptChecks          bool
-	EnableSyslog                bool
-	EnableUI                    bool
-	EncryptKey                  string
-	EncryptVerifyIncoming       bool
-	EncryptVerifyOutgoing       bool
-	HTTPAddrs                   []net.Addr
-	HTTPPort                    int
-	HTTPSAddrs                  []net.Addr
-	HTTPSPort                   int
-	KeyFile                     string
-	LeaveDrainTime              time.Duration
-	LeaveOnTerm                 bool
-	LogLevel                    string
-	NodeID                      types.NodeID
-	NodeMeta                    map[string]string
-	NonVotingServer             bool
-	PidFile                     string
-	RPCAdvertiseAddr            *net.TCPAddr
-	RPCBindAddr                 *net.TCPAddr
-	RPCHoldTimeout              time.Duration
-	RPCMaxBurst                 int
-	RPCProtocol                 int
-	RPCRateLimit                rate.Limit
-	RaftProtocol                int
-	ReconnectTimeoutLAN         time.Duration
-	ReconnectTimeoutWAN         time.Duration
-	RejoinAfterLeave            bool
-	RetryJoinIntervalLAN        time.Duration
-	RetryJoinIntervalWAN        time.Duration
-	RetryJoinLAN                []string
-	RetryJoinMaxAttemptsLAN     int
-	RetryJoinMaxAttemptsWAN     int
-	RetryJoinWAN                []string
-	SegmentName                 string
-	Segments                    []structs.NetworkSegment
-	SerfAdvertiseAddrLAN        *net.TCPAddr
-	SerfAdvertiseAddrWAN        *net.TCPAddr
-	SerfBindAddrLAN             *net.TCPAddr
-	SerfBindAddrWAN             *net.TCPAddr
-	SerfPortLAN                 int
-	SerfPortWAN                 int
-	ServerMode                  bool
-	ServerName                  string
-	ServerPort                  int
-	Services                    []*structs.ServiceDefinition
-	SessionTTLMin               time.Duration
-	SkipLeaveOnInt              bool
-	StartJoinAddrsLAN           []string
-	StartJoinAddrsWAN           []string
-	SyslogFacility              string
-	TLSCipherSuites             []uint16
-	TLSMinVersion               string
+
+	// Node name is the name we use to advertise. Defaults to hostname.
+	//
+	// NodeName is exposed via /v1/agent/self from here and
+	// used in lots of places like CLI commands. Treat this as an interface
+	// that must be stable.
+	// todo(fs): do we need this warning? RuntimeConfig was meant to be refactorable.
```

Comment:
This comment doesn't apply anymore now that we expose a `Config` member in /v1/agent/self that's not based on this structure. You can delete it.